gradle/prCheck.gradle

```diff
@@ -0,0 +1,95 @@
+        }
+    }
+}
+
+def checkPull(changeId) {
+    def jsonSlurper = new JsonSlurper()
+    def pullDetails = jsonSlurper.parseText(new URL("https://api.github.com/repos/h2oai/h2o-3/pulls/$changeId").getText())
+    def title = pullDetails.title
+    println ""
+    println "Checking pull request ($changeId): $title"
+    println ""
+    def hexdevMatcher = title =~ /.*HEXDEV-[0-9]{4,5}.*/
+    def itaMatcher = title =~ /.*ITA-[0-9]{3,5}.*/
+    if (hexdevMatcher.matches() || itaMatcher.matches()) {
+        println "HEXDEV and ITA jiras are not validated"
+        return []
+    }
+    def pubdevMatcher = title =~ /.*(PUBDEV-[0-9]{4,5}).*/
+    if (!pubdevMatcher.matches()) {
+        return ["Pull Request ${changeId} doesn't have associated JIRA ticket. Please add jira id in the title of the PR"]
+    }
+    def jiraId = pubdevMatcher[0][1]
+    def jiraDetails = jsonSlurper.parseText(new URL("https://h2oai.atlassian.net/rest/api/latest/issue/$jiraId").getText())
+    def status = jiraDetails.fields.status
+    if (status.name != 'In Progress') {
+        return ["Pull Request ${changeId} is linked to jira ticket https://h2oai.atlassian.net/browse/${jiraId}. " +
+                "This jira is not marked 'In Progress'. Please check if the associated jira ticket is correct and change the status to 'In Progress'"]
+    }
+    def fixVersions = jiraDetails.fields.fixVersions
+    if (fixVersions.size() == 0) {
+        return ["Pull Request ${changeId} is linked to jira ticket https://h2oai.atlassian.net/browse/${jiraId}. " +
+                "This jira doesn't have a fix version set. Please check if the associated jira ticket is correct and validate with the team what should be the correct fixed version."]
+    }
+    if (fixVersions.size() > 1) {
+        return ["Pull Request ${changeId} is linked to jira ticket https://h2oai.atlassian.net/browse/${jiraId}. " +
+                "This jira has more than one fix version. Please check if the associated jira ticket is correct and validate with the team what should be the correct fixed version."]
+    }
+    def fixVersionValue = fixVersions.get(0).name
+    def baseBranch = pullDetails.base.ref
+    if (fixVersionValue.endsWith('.1')) {
+        if (baseBranch != 'master')
+            return ["Pull Request ${changeId} is linked to jira ticket https://h2oai.atlassian.net/browse/${jiraId}. " +
+                    "This jira is marked for release in {$fixVersionValue}. This release will be done from a master branch. Change the base branch of the PR to 'master'."]
+    } else {
+        if (!baseBranch.startsWith('rel-')) {
+            return ["Pull Request ${changeId} is linked to jira ticket https://h2oai.atlassian.net/browse/${jiraId}. " +
+                    "This jira is marked for release in {$fixVersionValue}. This release will be done from a fix release branch. Change the base branch to a correspond to the right fix release branch."]
+        }
+    }
```

Comment:
Just point to think about. What about the task description? We can check if it is not empty, you can always write a link to the jira at least.

Reply:
I am not sure what you mean here. Task description where? Please elaborate

Reply:
Sorry, I meant PR description. We can check if the PR description is not empty.

Reply:
Yeah, I almost never manually fill the description. I put the important stuff in the commit messages (where it really matters). So I would be the victim of this check :)

Something to think about if this proves to be useful.

Reply:
:)